rstest_macros/src/parse/vlist.rs

```diff
@@ -39,25 +37,51 @@ pub(crate) struct ValueList {
     pub(crate) values: Vec<Value>,
 }
 
+/// Extracts the documentation comment from the attributes.
+fn extract_doc_comment(attrs: &[Attribute]) -> Option<String> {
+    attrs.iter().find_map(|attr| match &attr.meta {
+        Meta::NameValue(MetaNameValue {
+            path,
+            value: Expr::Lit(ExprLit {
+                lit: Lit::Str(doc), ..
+            }),
+            ..
+        }) if path.is_ident("doc") => Some(sanitize_ident(&doc.value().trim(), false)),
+        _ => None,
+    })
+}
+
 impl Parse for ValueList {
     fn parse(input: ParseStream) -> Result<Self> {
         let ident: Ident = input.parse()?;
         let _to: Token![=>] = input.parse()?;
         let content;
         let paren = syn::bracketed!(content in input);
-        let values: Expressions = content.parse()?;
 
-        let ret = Self {
-            arg: ident.into_pat(),
-            values: values.take().into_iter().map(|e| e.into()).collect(),
-        };
-        if ret.values.is_empty() {
+        let mut values = Vec::new();
+        while !content.is_empty() {
```

Comment:
syn proviide a macro to parse an homogeneous list of _parsabe_ items: see parse implementation in `Expressions`. Maybe create a new type `MatrixValue` and implement it that should parse `#[doc = ""..."]  #[doc = ""..."]  <Expression>`  salve also the previous issue.